src/main/java/com/powsybl/caseserver/datasource/S3CaseDataSourceService.java

```diff
@@ -51,34 +53,30 @@ public Boolean datasourceExists(UUID caseUuid, String fileName) {
 
     @Override
     public byte[] getInputStream(UUID caseUuid, String fileName) {
-        final var caseFileKey = Objects.nonNull(s3CaseService.getCompressionFormat(caseUuid)) && s3CaseService.getCompressionFormat(caseUuid).equals("zip")
-                ? uuidToPrefixKey(caseUuid) + fileName
-                : uuidToPrefixKey(caseUuid) + s3CaseService.getCaseName(caseUuid);
+        String caseName = s3CaseService.getCaseName(caseUuid);
+        String caseFileKey;
+        // For archived cases (.zip, .tar, ...), individual files are gzipped in S3 server.
+        // Here the requested file is decompressed and simply returned.
+        if (S3CaseService.isArchivedCaseFile(caseName)) {
+            caseFileKey = uuidToKeyWithFileName(caseUuid, fileName + GZIP_EXTENSION);
+            return s3CaseService.withS3DownloadedTempPath(caseUuid, caseFileKey,
+                    file -> S3CaseService.decompress(Files.readAllBytes(file)));
+        }
+        caseFileKey = uuidToKeyWithFileName(caseUuid, caseName);
         return withS3DownloadedDataSource(caseUuid, caseFileKey,
```

Comment:
should use withS3DownloadedTempPath as well ?

Reply:
Yes both withS3DownloadedDataSource can be deleted